MAVProxy/modules/mavproxy_link.py

```diff
@@ -110,23 +110,29 @@ def show_link(self):
             except AttributeError as e:
                 # some mav objects may not have a "signing" attribute
                 pass
-            print("link %u %s (%u packets, %.2fs delay, %u lost, %.1f%% loss%s)" % (master.linknum+1,
+            print("link %s %s (%u packets, %.2fs delay, %u lost, %.1f%% loss%s)" % (self.link_string(master),
                                                                                     status,
                                                                                     self.status.counters['MasterIn'][master.linknum],
                                                                                     linkdelay,
                                                                                     master.mav_loss,
                                                                                     master.packet_loss(),
                                                                                     sign_string))
+
     def cmd_link_list(self):
         '''list links'''
         print("%u links" % len(self.mpstate.mav_master))
         for i in range(len(self.mpstate.mav_master)):
             conn = self.mpstate.mav_master[i]
-            print("%u: %s" % (i, conn.address))
+            if conn.handle is not None:
+                print("%u (%s): %s" % (i, conn.handle, conn.address))
+            else:
+                print("%u: %s" % (i, conn.address))
 
-    def link_add(self, device):
+    def link_add(self, descriptor):
         '''add new link'''
         try:
+            link_components = descriptor.split('(')
```

Comment:
I wonder if we should do something more comprehensive in terms of adding attributes to links here?  Perhaps name-value pairs?

At the moment you're putting stuff in prens which is being dealt with by MAVProxy.  Another option would be to extend the link strings which pymavlink understands and have it remember the link names.  So your link might be:

`link add /dev/ttyUSB0:label=rfd900z`